Feat: Run OpenDistro without Security plugin

srajappa opened the pull request:
1. Updated `elasticsearch.yml` to disable opendistro security.
2. curl replaces wget in `securityadmin.sh` which runs in `config-*` pod.
3. Added another key in `config.json`; this way user has the ability to disable `xpack-security`.
4. `kibana_setup.sh` earlier required OpenDistro Security plugin. Now one has the choice to disable security.
5. Fixed few typos in README.md

Commit:
Feat: Run OpenDistro without Security plugin
1. Updated `elasticsearch.yml` to disable opendistro security.
2. curl replaces wget in `securityadmin.sh` which runs in `config-*` pod.
3. Added another key in `config.json`; this way user has the ability to disable `xpack-security`.
4. `kibana_setup.sh` earlier required OpenDistro Security plugin. Now one has the choice to disable security.
5. Fixed few typos in README.md

swoehrl-mw commented on elastic/universe/config.json:
```diff
@@ -842,6 +842,11 @@
           "type": "boolean",
           "default": true
         },
+        "xpack_security_enabled": {
```
This option doesn't do anything. It is not used, xpack is hard-coded as disabled (https://github.com/MaibornWolff/dcos-opendistro/blob/master/elastic/files/elasticsearch.yml#L44)

srajappa replied:
Agreed 👍 I will remove this from `config.json`, it is redundant and not useful much.

swoehrl-mw commented on kibana/files/kibana_setup.sh:
```diff
@@ -30,5 +30,11 @@ EOF
+if [ "$OPENDISTRO_SECURITY_ENABLED" = true ] then 
+	$MESOS_SANDBOX/kibana-$ELASTIC_VERSION-linux-x86_64/bin/kibana-plugin install file://$MESOS_SANDBOX/opendistro_security_kibana_plugin-${OPENDISTRO_SECURITY_VERSION}.zip
+else
+	sed -i '/^opendistro_security\.multitenancy\.enabled.*$/s/^/#/' $KIBANA_YAML
```
I think it would be better to move the entire block 
```
cat <<EOF >> $KIBANA_YAML
elasticsearch.username: $KIBANA_USER
elasticsearch.password: $KIBANA_PASSWORD
opendistro_security.multitenancy.enabled: false
EOF
```

into the if-then block instead of removing the line again later with sed.

srajappa replied:
Even I at the beginning had these bash statements in the `if-else-fi` block but I faced a problem below:

`Syntax error: end of file unexpected (expecting "fi")`

And something similar whenever I changed the indentation for those statements. 

EOF in an `if-else-fi` block was getting problematic, thus decided to have a `sed` statement in the end.

swoehrl-mw commented on kibana/universe/marathon.json.mustache:
```diff
@@ -49,8 +50,8 @@
-      "path": "/status",
-      "portIndex": 0,
+      "path": "/api/status",
+      "port": 5601,
```
Any particular reason why you want to switch from portIndex to port? I like portIndex better as to have fewer places where the actual port is used and therefore fewer places where it would need to be changed if one would want to do that.

srajappa replied:
The health-checks failed when using `portIndex`. When I changed it to the specific `port`, the health-checks succeeded.

swoehrl-mw commented on kibana/universe/marathon.json.mustache:
```diff
@@ -49,8 +50,8 @@
       "maxConsecutiveFailures": 4,
-      "path": "/status",
-      "portIndex": 0,
+      "path": "/api/status",
```
Good catch. Didn't realise in my testing that /status requires login and /api/status doesn't

Commit:
Fixes Some of the requested changes:
1. `xpack_security_enabled` block removed from `config.json`.
2. Unable to find the bash EOF block that works in an `if-else-fi` block.
3. Removed the `port` and readded `portIndex` in mustache file in `./kibana/universe`.
